### PR TITLE
{Test} Pipeline update

### PR DESCRIPTION
**Pull request by qwordy**
**Description**  

* Update pipeline run title.
* Generate index.html of testing results.
* Design a unique representation of a pipeline run.
* Make storage account container public.
* Fix a tiny DB bug.
* Fix pipe not close problem.

**Testing Guide**  

**History Notes**  

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.  
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

File: scripts/live_test/generate_index.py [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+    for blobs in root:
+        for blob in blobs:
+            name = url = ''
+            for e in blob:
+                if e.tag == 'Name':
+                    name = e.text
+                if e.tag == 'Url':
+                    url = e.text
+            if name == '' or url == '':
+                print('Warning, name: {}, url: {}'.format(name, url))
```
Comment (yungezz) [inline scripts/live_test/generate_index.py:36]: what's warning mean?

File: scripts/live_test/generate_index.py [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+        f.write(html)
+
+    # Upload to storage account
+    cmd = 'az storage blob upload -f index.html -c {} -n index.html --account-name clitestresultstac'.format(container)
+    print('Running: ' + cmd)
+    os.system(cmd)
+
+
+def render(data, container_url, testdata, USER_REPO, USER_BRANCH, COMMIT_ID, USER_LIVE):
+    content = """
```
Comment (yungezz) [inline scripts/live_test/generate_index.py:52]: how about extract it to a seperate file?

Comment (qwordy) [reply scripts/live_test/generate_index.py:52]: 144 lines. Short file. Don't worry.

File: scripts/live_test/generate_index.py [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+    </p>
+    """.format(USER_REPO, USER_BRANCH, COMMIT_ID, live)
+
+    content += """
+    <p>
+    <b>User Manual of Live Test Pipeline</b>
+    </p>
+    <p>
+    <a href=https://microsoft-my.sharepoint.com/:w:/p/fey/EZGC9LwrN3RAscVS5ylG4HMBX9h7W0ZSA7CDrhXN5Lvx6g?e=V8HUmd>Word</a> 
+    <a href=https://microsoft.sharepoint.com/teams/IoTToolingTeam/_layouts/OneNote.aspx?id=%2Fteams%2FIoTToolingTeam%2FShared%20Documents%2FAzure%20Management%20Experience%2FAzure%20Management%20Experience&wd=target%28AZ%20CLI%2FKnowledge%20base.one%7C18BC64EE-9328-497D-804E-6436006CA9A5%2FUser%20Manual%20of%20Live%20Test%20Pipeline%7C243EFA3E-FC7F-4612-9DA5-8E6BB2A11BD3%2F%29>OneNote</a>
```
Comment (fengzhou-msft) [inline scripts/live_test/generate_index.py:84]: Do these two docs have the same content? If so, you may remove the OneNote link as service team may not have permission to the OneNote page.

File: scripts/live_test/generate_index.py (hunk 2 above)
Comment (qwordy) [reply scripts/live_test/generate_index.py:52]: https://jinja.palletsprojects.com/en/2.11.x/ is a good choice. But it costs more time to write code. The current content is simple. If it gets complex enough, we can use template engine.

File: scripts/live_test/generate_index.py (hunk 1 above)
Comment (qwordy) [reply scripts/live_test/generate_index.py:36]: Let me compose more words.

File: scripts/live_test/generate_index.py (hunk 3 above)
Comment (qwordy) [reply scripts/live_test/generate_index.py:84]: Good idea.

**Review by fengzhou-msft — APPROVED** [review]

File: scripts/live_test/generate_index.py (hunk 3 above)
Comment (qwordy) [reply scripts/live_test/generate_index.py]: Done

File: scripts/live_test/generate_index.py (hunk 1 above)
Comment (qwordy) [reply scripts/live_test/generate_index.py]: Done

**Pull request merged**